### PR TITLE
Cc/salesforce webhook

### PR DESCRIPTION
**Pull request by cchrty**
### Briefly describe your changes:
Implemented PubSub API to subscribe to data change events on salesforce payment records

**Jira issue number and link:**
[PLTG-21](https://pltg.atlassian.net/browse/PLTG-21)
**Type of change:**
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Please answer the following:

**1. What is the current behavior?**
*If the feature did not exist, please say so.*
This is a new feature

**2. What is the new behavior?**
This webhook hook runs on a local server and receives data change events from Salesforce Payment records. based on the record type it will create a new invoice in stripe and send the stripe invoice ID back to the corresponding salesforce record.

**3. What tests did you run?**

### Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests for my code.
- [x] I have resolved all merge conflicts.

### CODE REVIEW

File: bill-bot-baggins/app/(main)/(routes)/payments/page.tsx
Comment (lhodges3) [inline bill-bot-baggins/app/(main)/(routes)/payments/page.tsx:1]: to be removed

File: bill-bot-baggins/app/api/auth/salesforce-auth/route.js
Comment (lhodges3) [inline bill-bot-baggins/app/api/auth/salesforce-auth/route.js:1]: to also be removed

**Review by lhodges3 — APPROVED** [review]
Looks good! Great work Chandler

**Review by Gambarou — APPROVED** [review]
Great work!

**Pull request merged**